src/work-allocation-2/containers/task-assignment/task-assignment-container.component.ts

```diff
@@ -47,8 +43,10 @@ export class TaskAssignmentContainerComponent implements OnInit, OnDestroy {
   private get returnUrl(): string {
     let url: string = '/work/my-work/list';
```

Comment:
this is still hard coded, is that correct?
how about doing a trick here?
if(verb ==='assign') {let url: string = '/work/all-work/list';}
else if(verb ==='reassign') {let url: string = '/work/my-work/list';}

Reply:
@rayliangatos - similar to above, it's hard-coded because the default return URL is the "My work" screen. It's not possible to set it based on the verb because, having discussed this with Tommy Wong (BA), it turns out the "All work" screen has both "assign" and "reassign" actions on tasks.

Reply:
I double check the code, actually this default value is useless, and it will be always overrided by:
`    if (window && window.history && window.history.state && window.history.state.returnUrl) {
      url = window.history.state.returnUrl;
    }`

In that case could you assign the empty value to it? like:
```suggestion
    let url: string = '';
```
Then it will be a genuine generic component.

Reply:
Not always, @rayliangatos... I came across a case where `returnUrl` is `undefined` if the user uses the browser navigation buttons to go back and then forwards - hence I had to check for `window.history.state.returnUrl`.

Reply:
We can't control user and prohibit them to use the back button in browser, but in that case if user use the Back button in browser when in All Work->tasks, this cancel button will go back to My Work->my tasks, that shouldn't be allowed, we would rather let the user go back to their home page, that is '' for them. As not everyone can access both My Work and ALL Work. I would give you a demo but hard code as '/work/my-work/list' this cause an edge case issue in All Work.

Reply:
Ah... I didn't know that - I thought everyone who had access to "All work" also had "My work" but this appears not to be the case. I will default it to `''`, then. I think there is also another location where I will have to change it to `''`.